### ISSUE TITLE
Atrributes and Variations Lost On Update

### ISSUE THREAD

**Issue by sellitto** [woocommerce/woocommerce#14881]
### EXPLANATION OF THE ISSUE

I import my products from a third party, this was working awesome until my site auto updated on April 21st, to current version of WordPress and Woocommerce. Once products are imported I need edit each one and assign category, update description, etc., When I do this and save, all attributes and variations are being lost. I have lost A LOT of work since the 21st because of this. I have disabled all plugins and switched to the Twenty Twelve theme and the issue still exists.  I am loosing business due to this and I did everything on my end to resolve, even upgrading to PHP 7+ and my MySQL database to 5.6. 

### STEPS TO REPRODUCE THE ISSUE

I do not add products to my site as they are imported, the third party has said they compliant with new Woocommerce 3.0 changes. If you view an imported product and its details everything is in tact and correct, attributes and variations until I edit after import and the product looses it attributes and variations. They all have same the same attributes of size and color. I did read another issue with similar issues but saw no resolution and I need one asap.

### SYSTEM STATUS

<details>
```
`
### WordPress Environment ###

Home URL: http://products.poshdesign.co
Site URL: http://products.poshdesign.co
WC Version: 3.0.5
Log Directory Writable: ✔
WP Version: 4.7.4
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US

### Server Environment ###

Server Info: Apache/2.2.24 (Unix) mod_hive/5.5 mod_ssl/2.2.24 OpenSSL/1.0.0-fips mod_auth_passthrough/2.1 mod_bwlimited/1.4 mod_fastcgi/2.4.6 mod_fcgid/2.3.6
PHP Version: 7.0.16
PHP Post Max Size: 8 MB
PHP Time Limit: 3000
PHP Max Input Vars: 2000
cURL Version: 7.53.0
OpenSSL/1.0.2k

SUHOSIN Installed: –
MySQL Version: 5.6.35
Max Upload Size: 2 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.0.5
WC Database Prefix: keto_
woocommerce_sessions: ✔
woocommerce_api_keys: ✔
woocommerce_attribute_taxonomies: ✔
woocommerce_downloadable_product_permissions: ✔
woocommerce_order_items: ✔
woocommerce_order_itemmeta: ✔
woocommerce_tax_rates: ✔
woocommerce_tax_rate_locations: ✔
woocommerce_shipping_zones: ✔
woocommerce_shipping_zone_locations: ✔
woocommerce_shipping_zone_methods: ✔
woocommerce_payment_tokens: ✔
woocommerce_payment_tokenmeta: ✔

### Security ###

Secure connection (HTTPS): ❌Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ✔

### Active Plugins (21) ###

Admin Columns Pro: by AdminColumns.com – 4.0.4
Advanced Custom Fields PRO: by Elliot Condon – 5.5.12
Auto Terms of Service and Privacy Policy: by TourKick (Clifford P) – 1.8
Admin Columns - Advanced Custom Fields add-on: by Codepress – 2.0.3
Admin Columns - WooCommerce add-on: by Codepress – 2.1.2
Contact Form 7: by Takayuki Miyoshi – 4.7
EWWW Image Optimizer: by Shane Bishop – 3.3.0
Loginizer: by Raj Kothari – 1.3.4
Nextend Facebook Connect: by Nextendweb – 2.0.2
Nextend Google Connect: by Roland Soos
Jamie Bainbridge – 1.6.1

Regenerate Thumbnails: by Alex Mills (Viper007Bond) – 2.2.6
UpdraftPlus - Backup/Restore: by UpdraftPlus.Com
DavidAnderson – 2.11.17.1

User Role Editor: by Vladimir Garagulya – 4.32.3
Woocommerce Advanced Bulk Edit: by George Iron – 4.2
WooCommerce Advanced Shipping: by Jeroen Sormani – 1.0.13
WooCommerce: by Automattic – 3.0.5
WooSidebars: by WooThemes – 1.4.3
WP-Lister Lite for Amazon: by Matthias Krok – 0.9.6.33
WP-Lister Lite for eBay: by Matthias Krok – 2.0.17
WP Security Audit Log: by WP White Security – 2.6.3
YITH WooCommerce Wishlist: by YITHEMES – 2.1.1

### Settings ###

API Enabled: ✔
Force SSL: ✔
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)


### WC Pages ###

Shop base: #3 - /
Cart: #4 - /cart/
Checkout: #5 - /checkout/
My account: #6 - /my-account/

### Theme ###

Name: Flatsome Child
Version: 3.0
Author URL: 
Child Theme: ✔
Parent Theme Name: Flatsome
Parent Theme Version: 3.3.5
Parent Theme Author URL: http://www.uxthemes.com/
WooCommerce Support: ✔

### Templates ###

Overrides: flatsome/woocommerce/archive-product.php
flatsome/woocommerce/cart/cart-empty.php
flatsome/woocommerce/cart/cart.php
flatsome/woocommerce/checkout/form-checkout.php
flatsome/woocommerce/checkout/form-coupon.php
flatsome/woocommerce/checkout/thankyou.php
flatsome/woocommerce/content-product.php
flatsome/woocommerce/content-product_cat.php
flatsome/woocommerce/content-single-product.php
flatsome/woocommerce/global/breadcrumb.php
flatsome/woocommerce/global/quantity-input.php
flatsome/woocommerce/global/wrapper-end.php
flatsome/woocommerce/global/wrapper-start.php
flatsome/woocommerce/loop/loop-end.php
flatsome/woocommerce/loop/loop-start.php
flatsome/woocommerce/loop/orderby.php
flatsome/woocommerce/loop/pagination.php
flatsome/woocommerce/loop/result-count.php
flatsome/woocommerce/loop/sale-flash.php
flatsome/woocommerce/myaccount/form-login.php
flatsome/woocommerce/myaccount/navigation.php
flatsome/woocommerce/notices/error.php
flatsome/woocommerce/notices/notice.php
flatsome/woocommerce/notices/success.php
flatsome/woocommerce/product-searchform.php
flatsome/woocommerce/single-product/photoswipe.php
flatsome/woocommerce/single-product/price.php
flatsome/woocommerce/single-product/product-image.php
flatsome/woocommerce/single-product/product-thumbnails.php
flatsome/woocommerce/single-product/related.php
flatsome/woocommerce/single-product/review.php
flatsome/woocommerce/single-product/sale-flash.php
flatsome/woocommerce/single-product/share.php
flatsome/woocommerce/single-product/short-description.php
flatsome/woocommerce/single-product/tabs/tabs.php
flatsome/woocommerce/single-product/title.php
flatsome/woocommerce/single-product/up-sells.php
flatsome/woocommerce/single-product-reviews.php
flatsome/woocommerce/single-product.php

`
```
</details>

**Comment by vallter2** [comment]
Hi,
i have same problem. Variations is not visible in edit post, but on frontend is visible, but I can't Add to Cart. If update post variations lost... When create new product all work fine. 

Template is compatible with Woocommerce (new version 3.0.5) - betheme.

I'm try to disable all plugin  but same problem...

**Comment by claudiulodro** [comment]
@sellito, I believe the problems you are experiencing come from a bug in attributes first documented in #14824. This should be fixed in 3.0.6, which is releasing shortly.

@vallter2, your problem sounds different. Can you please make a new issue documenting it with steps to reproduce?

**Comment by sellitto** [comment]
Thanks for the response, by chance could this also be causing variation
image for color variation not too be saved when a new product is added, or
not possible at all?
